cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/inference/Inference.kt

```diff
@@ -605,3 +667,197 @@ fun RecordDeclaration.inferMethod(
             call
         ) as? MethodDeclaration
 }
+
+fun TranslationContext.tryNamespaceInference(
+    name: Name,
+    locationHint: Node?
+): NamespaceDeclaration? {
+    // Determine the scope where we want to start our inference
+    var (scope, _) = scopeManager.extractScope(name, location = locationHint?.location)
+
+    if (scope !is NameScope) {
+        scope = null
+    }
+
+    var holder = scope?.astNode
+
+    // If we could not find a scope, but we have an FQN, we can try to infer a namespace (or a
+    // parent record)
+    var parentName = name.parent
+    if (scope == null && parentName != null) {
+        holder = tryParentScopeInference(parentName, locationHint)
+    }
+
+    return (holder ?: scopeManager.globalScope?.astNode)
+        ?.startInference(this)
+        ?.inferNamespaceDeclaration(name, null, locationHint)
+}
+
+/**
+ * Tries to infer a [RecordDeclaration] from an unresolved [Type]. This will return `null`, if
+ * inference was not possible, or if it was turned off in the [InferenceConfiguration].
+ */
+fun TranslationContext.tryRecordInference(
```

Comment:
add a quick comment that we are also inferring the (nested) scope of the record, if it does not exist